app/viewmodels/SecuritiesRequestedStatementsViewModel.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import helpers.Formatters.dateAsDayMonthAndYear
+import models.FileFormat.Pdf
+import models.SecurityStatementsForEori
+import play.api.i18n.Messages
+import play.twirl.api.{Html, HtmlFormat}
+
+case class StatementRow(historyIndex: Int,
+                        eori: Option[String],
+                        date: String,
+                        pdf: Option[PdfLink],
+                        rowId: String,
+                        dateCellId: String,
+                        linkCellId: String)
+
+case class PdfLink(downloadURL: String, formattedSize: String, ariaText: String)
+
+object SecuritiesRequestedStatementsViewModel {
+  def prepareStatementRows(securityStatements: Seq[SecurityStatementsForEori])
+                          (implicit messages: Messages): Seq[StatementRow] = {
+
+    securityStatements.zipWithIndex.flatMap { case (statement, statementIndex) =>
+      statement.requestedStatements.sorted.zipWithIndex.map { case (requestedStatement, requestedIndex) =>
+
+        val eori = if (statementIndex > 0) Some(statement.eoriHistory.eori) else None
```

Comment:
Better to move the code from line 42 -72 in a separate method with meaningful name in order to make it more readable